### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by s-taylor**
This PR adds the ability to scaffold a user model via `redbeard user [name]`
name is optional and defaults to "user" but this could be changed to "admin" or something similar.

The user scaffolding still utilises the controller scaffold as there was a huge amount of code overlap between the two but using a number of conditionals adds extra routes and makes changes where required. This was discussed with Dave and if any more conditionals are required in controllers a better solution is required.

In addition...
- we are now using the `standard-http-error` module to bubble status codes from the controller to routes.
- a few of the standard controller tests needed to be changed to support users, the main issue was when creating or deleting you provide a password but this gets converted to a hash, so object comparisons need to strip the password for this to work.

Note: This is my first attempt ever at implementing auth so I probably missed things, please let me know what these things are! I haven't implemented any middleware for route filtering yet based on user signin. This can be a separate PR.

### CODE REVIEW

**Comment by s-taylor** [comment]
Also, one issue i've run into is there are some additional npm modules required for users to function. The problem is package.json is generated when base is run, and we can't edit it when the user then later runs `redbeard user`. What this means is that there may be unnecessary packages in package.json but i'm not sure how we can actually fix this problem?

**Comment by nwinch** [comment]
> Also, one issue i've run into is there are some additional npm modules required for users to function. The problem is package.json is generated when base is run, and we can't edit it when the user then later runs redbeard user. What this means is that there may be unnecessary packages in package.json but i'm not sure how we can actually fix this problem?

I have two suggestions:

1. Install the needed packages with `child_process` - similar to what's happening in `./test.js`
2. We could do this dynamically using `npm`s API via require:
```
var npm = require('npm');

// commands to do things
npm.load
npm.commands.install
```

But this isn't a recommended - as in not supported, and could change - thing to do with `npm`, and rather use it via CLI.  But, it does remain an option :)

File: base/package.mustache [hunk 1]
```diff
@@ -16,10 +16,12 @@
     "json-schema-deref-sync": "^0.2.8",
     "jsonwebtoken": "^5.4.1",
     "lodash": "^3.10.1",
+    "password": "https://github.com/Prismatik/password.git",
```
Comment (nwinch) [inline base/package.mustache]: You'll need the `git` prefix as well: `git+https://github.com/Prismatik/password.git`

File: controller/controller.mustache [hunk 2]
```diff
@@ -3,14 +3,19 @@ const r = require('../lib/db');
...
 const modelSchema = appSchema['{{name}}'];
 
-const LINK = 'https://rethinkdb.com/docs/changefeeds/javascript/'
-var RESTRICTED_FEED = "WARNING: Changefeed is restricted and will only"
-RESTRICTED_FEED += " include updates to existing documents, not newly created"
-RESTRICTED_FEED += " documents. Read more here: " + LINK;
+const LINK = 'https://rethinkdb.com/docs/changefeeds/javascript/';
+var RESTRICTED_FEED = 'WARNING: Changefeed is restricted and will only';
+RESTRICTED_FEED += ' include updates to existing documents, not newly created';
+RESTRICTED_FEED += ' documents. Read more here: ' + LINK;
```
Comment (nwinch) [inline controller/controller.mustache]: Do we have many 'warnings' ?  If we have more than one, I think we could move these to a `/constants` folder or similar, so as not to clog up space.

File: package.json [hunk 3]
```diff
@@ -35,10 +35,12 @@
     "mocha": "^2.4.5",
     "must": "^0.13.1",
     "node-fetch": "^1.3.3",
+    "password": "https://github.com/Prismatik/password.git",
```
Comment (nwinch) [inline package.json]: Add `git+` prefix

**Comment by nwinch** [comment]
@s-taylor Review done with a couple of comments - I have some more, but they're related to `redbeard` in general so I'm going to post them as separate discussion/issues

**Comment by davidbanham** [comment]
re: the package.json thing, it's just json. Nothing stopping us from:

```
const package = fs.readFileSync('package.json');
const updated = _.merge(package.dependencies, newDeps);
fs.writeFileSync('package.json', updated);
```

I had hoped to avoid things ever needing to share files, but maybe that's a futile dream.

I am painfully aware of #66 at this moment.

**Comment by s-taylor** [comment]
I've gone with @davidbanham 's suggestion, @nwinch do you want to review or happy to merge?

**Comment by nwinch** [comment]
Yep cool

File: lib/package.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const fs = require('fs');
+const _ = require("lodash");
+
+// add dependencies to an already generated package.json file
+exports.addDependencies = function(newDependencies) {
+  const packagePath = process.cwd() + '/package.json';
+  const package = JSON.parse(fs.readFileSync(packagePath));
+  const updated = _.merge(package, {dependencies: newDependencies});
+  fs.writeFileSync(packagePath, JSON.stringify(updated, null, 2));
+};
```
Comment (nwinch) [inline lib/package.js]: Now that this is a separate it makes me feel it should have some tests.  Thoughts?

Comment (s-taylor) [inline lib/package.js]: Yea probably. It uses process.cwd() which makes that trickier but I guess i can just mock it?

**Comment by s-taylor** [comment]
Hold off merging, Travis broken so I need to figure out what's going on...

**Pull request merged**